src/memkind_memtier_mtt_allocator.c

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/* Copyright (C) 2022 Intel Corporation. */
+
+#include <memkind/internal/memkind_memtier_mtt_allocator.h>
+#include <memkind/internal/memkind_private.h>
+
+pthread_t bg_thread;
```

Comment:
Maybe it would be better to make it a field in a structure passed to create and destroy functions?

We will need to keep more data there...

```c
typedef struct {
    // in the future, this structure will contain more fields
    pthread_t bg_thread;
} MTTAllocator;
```


This should also be defined in memkind_memtier_mtt_allocator.h, having instance is not required for now - we can test it using UT